Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py

```diff
@@ -169,10 +175,30 @@ def main(cls, args, repository, why=None, redact=False, **kwargs):
             sys.stderr.write("'{}' is an invalid branch name, cannot create it\n".format(args.issue))
             return 1
 
-        remote_re = re.compile('remotes/.+/{}'.format(re.escape(args.issue)))
-        for branch in repository.branches:
-            if branch == args.issue or remote_re.match(branch):
-                sys.stderr.write("'{}' already exists\n".format(args.issue))
+        if args.issue in repository.branches_for(remote=target_remote):
+            if not args.delete_existing:
+                sys.stderr.write("'{}' exists on the remote '{}' and will be overwritten by a push\n".format(args.issue, target_remote))
+                if args.delete_existing is False:
+                    return 1
+
+        for remote in repository.source_remotes():
+            if args.issue in repository.branches_for(remote=remote):
+                sys.stderr.write("'{}' cannot be used as a PR branch, it exists on remote '{}'\n".format(args.issue, remote))
+                return 1
```

Comment:
What is this guarding against? I can't think of a reason why we'd compute a branch name that happens to match a branch name on origin, but that shouldn't _technically_ be unusable as a PR branch, right?

Even `main` is a valid PR branch!

Reply:
(Well, maybe a PR from `fork/main` would break EWS, but at least it's valid for Git and GitHub 😛)

Reply:
I was guarding against things like `fork/main` and `fork/safari-613-branch`, but this is probably more trouble than it's worth the more I think about it, will remove this code.